### PR TITLE
Nginx public_*html permission error fix

### PR DESCRIPTION
**Pull request by drsdre**
The Nginx error logs are getting filled up with:
`2020/06/18 11:41:09 [crit] 12961#12961: *43239 open() "/home/user/web/website.com/public_html/" failed (13: Permission denied), client: 108.61.255.255, server: website.com, request: "GET / HTTP/1.1", host: "website.com"
`

This is caused by too tight permissions on the /home/$user/web/$domain/public_html (and public_shtml) directory for the www-data user. The current permission is 751 for user.user. As www-data is not part of the 'user' group, it cannot access the directory itself. With this pull request I propose to use the 755 permissions for public_*html which means everyone can access the directory including www-data.

Is there a security risk to this? It's not likely. Higher up directories like /home/user/web/website.com are set to no allow other users to read which means they can also not access /home/user/web/website.com/public_html

### CODE REVIEW

File: bin/v-add-web-domain [hunk 1]
```diff
@@ -94,6 +94,7 @@ chown root:$user /var/log/$WEB_SYSTEM/domains/$domain.* $conf
 chmod 640 /var/log/$WEB_SYSTEM/domains/$domain.*
 chmod 751 $HOMEDIR/$user/web/$domain $HOMEDIR/$user/web/$domain/*
 chmod 551 $HOMEDIR/$user/web/$domain/stats $HOMEDIR/$user/web/$domain/logs
+chmod 755 $HOMEDIR/$user/web/public_*html
```
Comment (jaapmarcus) [inline bin/v-add-web-domain]: Shouldn't this not be  $HOMEDIR/$user/web/$domain/public_*html ?

$HOMEDIR/$user/web/public_*html does not exists...

**Comment by drsdre** [comment]
Thanks @jaapmarcus well spotted. Fixed with latest commit.

**Comment by Lupul** [comment]
Related Issue: #792

File: func/rebuild.sh [hunk 2]
```diff
@@ -98,6 +98,7 @@ rebuild_user_conf() {
         mkdir -p $HOMEDIR/$user/tmp
         chmod 751 $HOMEDIR/$user/conf/web
         chmod 751 $HOMEDIR/$user/web
+        chmod 755 $HOMEDIR/$user/web/public_*html
```
Comment (Lupul) [inline func/rebuild.sh]: No need for this, public _html is only present in domain folder

Comment (drsdre) [reply func/rebuild.sh]: You are correct. In the v-rebuild-web-domains the correct permissions are set.

**Comment by Lupul** [comment]
@drsdre can you please check if changing group ownership fixes this issue
> chown $user:www-data $HOMEDIR/$user/web/$domain/public_*html

**Comment by ioannidesalex** [comment]
If 755 it should be user:user and errors are gone.

Didn’t try 751 with user:www-data though

**Comment by drsdre** [comment]
As expected user:www-data also solves the permission errors.

Then I think we should go for user:www-data as that opens up the directory less than giving others read permission.

If agreed, I'll update the Pull Request.

**Comment by ScIT-Raphael** [comment]
@drsdre As I can see in our internal team chat, @jaapmarcus and @lupul are already working on a fix, the incoming night just stopped them :). Please give us around 12hrs to see where the plan goes, but will be probaly a chown $user:www-data.

Thanks for your contribution, you made it possible to discuss the issue and find possible fixes!!!

**Comment by drsdre** [comment]
Thanks @ScIT-Raphael

**Comment by ScIT-Raphael** [comment]
PS: Would you like to get in touch with the team? We always search new dev members, infact you helped us already a lot! Just raise a mail to info@hestiacp.com, I would love to send you additional informations.

**Pull request closed without merging**